app/views/user_sessions/_new.html.erb

```diff
@@ -38,4 +38,15 @@
     <p style="color: #888;"><%= raw t('user_sessions.new.reset_by_clicking_here', :url1 => "/reset/") %></p>
   </div>
   <% end %>
+  <% if !current_user %>
+    <script>
+      $(document).ready(function() {
+        $('a.requireLogin').on('click', function(e) {
+          e.preventDefault();
+          $('#loginModal').modal();
+          $.getJSON($(this).attr('href'));
```

Comment:
Hmm, what does this line do exactly? I wonder if we should explain in a comment so people understand this in the future. Maybe also add a comment above this code segment to tell what it does so people are aware in the future and don't mess it up? Just thinking about how critical this is!

Reply:
Thanks Jeff for reminding me to guide students about documentations. :-)